src/intfuncs.c

```diff
@@ -461,7 +462,7 @@ Obj FuncHASHKEY_BAG(Obj self, Obj obj, Obj seed, Obj offset, Obj maxlen)
       n = imaxlen;
   }
 
-  return INTOBJ_INT(HASHKEY_BAG_NC(obj, (UInt4)s, offs, (int)n));
+  return ObjInt_Int(HASHKEY_BAG_NC(obj, (UInt4)s, offs, (int)n));
```

Comment:
But we deliberately wrote `HASHKEY_MEM_NC` (and thus code using it, like `HASHKEY_BAG_NC`) to return an integer that fits into an immediate integer object. And there is code that explicitly relies on `HASHKEY_BAG` returning an immediate integer value. I feel this change obscures this. At the very least, we'd have to do somthing lke this:
```C
  Obj result = ObjInt_Int(HASHKEY_BAG_NC(...));
  if (!IS_INTOBJ(result)) Panic("this must never happen");
  return result;
```
But it seems better to me to just leave it alone, but add comment saying something like
```C
// HASHKEY_BAG_NC is guaranteed to return a value that fits into an immediate integer
```
and then perhaps we should also insert such comments into `HASHKEY_BAG_NC` and `HASHKEY_MEM_NC`

Reply:
I agree this makes it harder to read, but I'd also feel happier if we checked the answer did fit in an immediate integer (and in this case, I don't think the overhead is painful). I could look again into a CheckedINTOBJ_INT

Reply:
I don't mind the overhead of the check; I mind that we first go to great length to create a value that fits, and then apply a bludgeon to it "just in case". If we are concerned that we got the code to create the values wrong, then we should add assertions and/or `if (...) Panic(...)` checks there, right after that. Then change `HASHKEY_MEM_NC` and `HASHKEY_BAG_NC` to return an immediate integer as `Obj`. Then adapt the few callers to that (it's trivial in all cases, I think -- no need to convert back there with `INT_INTOBJ` or `Int_ObjInt`)... except that unfortunately, the `datastructures` package uses `HASHKEY_MEM_NC` and `HASHKEY_WHOLE_BAG_NC`, so we can't just change the API *sigh*. But we could "deprecate" them, and add new variants which return `Obj`...